components/search_results/search_limits_banner.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import styled from 'styled-components';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {isAdmin} from 'mattermost-redux/utils/user_utils';
+import {isCurrentLicenseCloud, getSubscriptionProduct as selectSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {CloudProducts} from 'utils/constants';
+import useGetUsage from 'components/common/hooks/useGetUsage';
+import {asGBString, fallbackStarterLimits} from 'utils/limits';
+
+const StyledDiv = styled.div`
+width: 100%;
+`;
+
+const StyledA = styled.a`
+color: rgba(var(--denim-button-bg-rgb, 1)) !important;
+`;
+
+const InnerDiv = styled.div`
+display: flex;
+gap: 8px;
+border-radius: 4px;
+border: 1px solid rgba(var(--center-channel-text-rgb), 0.08);
+border-radius: 4px;
+background-color: rgba(var(--center-channel-text-rgb), 0.04);
+padding: 10px;
+margin: 10px;
+color: rgba(var(--center-channel-text-rgb), 0.72);
+font-weight: 400;
+font-size: 11px;
+line-height: 16px;
+letter-spacing: 0.02em;
+`;
+
+type Props = {
+    searchType: string;
+}
+
+const FILES_SEARCH_TYPE = 'files';
+const MESSAGES_SEARCH_TYPE = 'messages';
+
+function SearchLimitsBanner(props: Props) {
+    const {formatMessage, formatNumber} = useIntl();
+    const openPricingModal = useOpenPricingModal();
+    const usage = useGetUsage();
+    const isAdminUser = isAdmin(useSelector(getCurrentUser).roles);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const product = useSelector(selectSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+
+    if (!isCloud || !isStarter) {
+        return null;
+    }
+
+    const currentFileStorageUsage = usage?.files?.totalStorage || 0;
+    const currentMessagesUsage = usage?.messages?.history || 0;
+
+    let ctaAction = formatMessage({
+        id: 'workspace_limits.search_limit.view_plans',
+        defaultMessage: 'View plans',
+    });
+
+    if (isAdminUser) {
+        ctaAction = formatMessage({
+            id: 'workspace_limits.search_limit.upgrade_now',
+            defaultMessage: 'Upgrade now',
+        });
+    }
+
+    const renderBanner = (bannerText: React.ReactNode, id: string) => {
+        return (<StyledDiv id={id}>
+            <InnerDiv>
+                <i className='icon-eye-off-outline'/>
+                <span>{bannerText}</span>
+            </InnerDiv>
+        </StyledDiv>);
+    };
+
+    switch (props.searchType) {
+    case FILES_SEARCH_TYPE:
+        if (!(currentFileStorageUsage > fallbackStarterLimits.files.totalStorage)) {
```

Comment:
These should use the current subscription limits fetched from CWS. i.e. the ones fetched by using [`useGetLimits`](https://github.com/mattermost/mattermost-webapp/blob/master/components/common/hooks/useGetLimits.ts#L12). The fallbackStarterLimits are only applicable for static things not linked to the *current* subscription, like showing what the limits are for starter plan in the pricing modal.